backend/app/api/users.py

```
@@ -0,0 +1,248 @@
+            "profile_picture_url": profile_picture_url
+        }
+        
+    except Exception as e:
+        # Clean up file if something went wrong
+        if os.path.exists(file_path):
+            os.remove(file_path)
+        
+        raise HTTPException(
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            detail="Failed to process image"
+        )
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)